System.IO.FileSystem/FileSystemManager.cs

```diff
@@ -0,0 +1,263 @@
+﻿//
+// Copyright (c) .NET Foundation and Contributors
+// See LICENSE file in the project root for full license information.
+//
+
+using System.Collections;
+
+namespace System.IO
+{
+    internal class FileSystemManager
```

Comment:
should this one be disposable or have a mechanism to close the files open or anything that will prevent a file to be still open?
We could use a create factory for that one and then at dispose, it will properly close all the native file? Something along those lines?

Reply:
This is already handled by this class along with the removable events handlers and HAL init/deinit.